src/pint/pint_matrix.py

```diff
@@ -72,16 +79,65 @@ def label_units(self):
             units.append(self.get_axis_labels(dim))
         return units
 
-    def get_label_size(self, label):
-        """Get the size of the a label in each axis.
+    def get_label_names(self, axis=None):
+        """return only the names of the labels
+        along the specified axis if requested
+
+        Parameters
+        ----------
+        axis: int (optional)
+            axis for the label lookup
+
+        Returns
+        -------
+        labels : list of label names for each dimension
+
+        """
+        labels = []
+        if axis is None:
+            r = range(len(self.axis_labels))
+        else:
+            if isinstance(axis, int):
+                return [x[0] for x in self.get_axis_labels(axis)]
+            else:
+                # assume it's an iterable (how to check?)
+                r = axis
+        for dim in r:
+            labels.append([x[0] for x in self.get_axis_labels(dim)])
+        return labels
+
+    def get_all_label_names(self):
```

Comment:
I just worry that general deduplication will look like it's going to give just one copy for covariance matrices, until someone runs into a covariance matrix with repeated labels on the same axis, when it's going to do something different from what they assume.

Reply:
I think this facility is important because this works for other matrices as well which might have repeated labels.  There are differences between what I did (relating to parameter covariance matrices) and other sorts (data covariance matrices, ...).  This may be historical - I don't know how much this is used.

So I would be fine with changing this behavior but not without understanding what other uses are.

Alternately, could make a subclass that does not allow for repeated labels.